### PR TITLE
Fixes #8785: Improve Script field behavior in passwords

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/8785

### CODE REVIEW

File: rudder-web/src/main/webapp/javascript/rudder/angular/passwordForm.js [hunk 1]
```diff
@@ -52,38 +52,46 @@ passwordModule.controller("passwordController", function($scope) {
...
+  updateResult();
 
   $scope.$watch('action',updateResult);
   $scope.$watch('newPassword.password',updateResult);
   $scope.$watch('newPassword.hash',updateResult);
 
   function updateResult () {
     // Keep and delete, use current password as base
-    var result = $scope.current
+    var result = angular.copy($scope.current)
```
Comment (ncharles) [inline rudder-web/src/main/webapp/javascript/rudder/angular/passwordForm.js]: missing ; there

File: rudder-web/src/main/webapp/javascript/rudder/angular/passwordForm.js [hunk 2]
```diff
@@ -52,38 +52,46 @@ passwordModule.controller("passwordController", function($scope) {
...
     if ($scope.action === "change") {
-      result = $scope.newPassword;
+      result = angular.copy($scope.newPassword);
     }
+    
+    if (result.hash === "plain" && result.isScript) {
+      result.password = "evaljs:"+result.password
```
Comment (ncharles) [inline rudder-web/src/main/webapp/javascript/rudder/angular/passwordForm.js]: missing ; there

**Comment by ncharles** [comment]
Appart from the two missing ;, this is awesome

**Comment by ncharles** [comment]
I've been testing it, and when we fill a password value, then click "add another", the field is emptied (looks like all the password fields get back their original value in this case)
Please fix this before we can merge this PR

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by ncharles** [comment]
Awesome!

**Comment by ncharles** [comment]
This is good for me (except the missing ; )
so you can add them, and self merge

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**